linter/lintfuncs_test.go

```diff
@@ -21,18 +21,18 @@ func TestLintProducedAtDate(t *testing.T) {
 	}
 
 	t.Run("Old ProducedAt date", func(t *testing.T) {
-		err = LintProducedAtDate(ocspResp)
-		if err == nil {
-			t.Errorf("Should have had error: %s is more than %s in the past", ocspResp.ProducedAt.String(), ProducedAtLimit)
+		status, info := LintProducedAtDate(ocspResp, nil)
+		if status != Failed {
+			t.Errorf("Should have had error, instead got: %s", info)
 		}
 	})
 
 	ocspResp.ProducedAt = time.Now()
 
 	t.Run("Happy path", func(t *testing.T) {
-		err = LintProducedAtDate(ocspResp)
-		if err != nil {
-			t.Errorf("Should not have gotten error, instead got error: %s", err.Error())
+		status, info := LintProducedAtDate(ocspResp, nil)
+		if status != Passed {
```

Comment:
Similarily here it looks like the condition should be `status==Error`

Reply:
status can be `Passed`, `Failed`, or `Error` so anything other than `Passed` needs to lead to an error

Reply:
Yeah that's definitely a good idea